src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,132 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int capasity;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        capasity = INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (mapIsLoad()) {
+            resize();
+        }
+        table = putElement(key, value, table);
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> curentNode = table[getIndex(key)];
+        while (curentNode != null) {
+            if (keyCompare(curentNode.key, key)) {
+                return curentNode.value;
+            }
+            curentNode = curentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[capasity << 1];
+        capasity = capasity << 1;
+        size = 0;
+        for (int i = 0; i < table.length; i++) {
+            if (table[i] == null) {
+                continue;
+            }
```

Comment:
this check is unnecessary, cause if `table[i] = null` while loop won't work anyway and you just go to the next table item